pyccel/ast/cudaext.py

```diff
@@ -0,0 +1,209 @@
+from functools import reduce
+import operator
+
+import numpy
+
+from pyccel.errors.errors import Errors
+from pyccel.errors.messages import WRONG_LINSPACE_ENDPOINT, NON_LITERAL_KEEP_DIMS, NON_LITERAL_AXIS
+
+from pyccel.utilities.stage import PyccelStage
+
+from .basic          import PyccelAstNode
+from .builtins       import (PythonInt, PythonBool, PythonFloat, PythonTuple,
+                             PythonComplex, PythonReal, PythonImag, PythonList,
+                             PythonType, PythonConjugate)
+
+from .core           import process_shape, Module, Import, PyccelFunctionDef
+
+from .datatypes      import (dtype_and_precision_registry as dtype_registry,
+                             default_precision, datatype, NativeInteger,
+                             NativeFloat, NativeComplex, NativeBool, str_dtype,
+                             NativeNumeric)
+
+from .internals      import PyccelInternalFunction, Slice, max_precision, get_final_precision
+from .internals      import PyccelArraySize
+
+from .literals       import LiteralInteger, LiteralFloat, LiteralComplex, convert_to_literal
+from .literals       import LiteralTrue, LiteralFalse
+from .literals       import Nil
+from .mathext        import MathCeil
+from .operators      import broadcast, PyccelMinus, PyccelDiv
+from .variable       import (Variable, Constant, HomogeneousTupleVariable)
+
+from .numpyext       import process_dtype, NumpyNewArray
+
+#==============================================================================
+__all__ = (
+    'CudaMemCopy',
+    'CudaNewArray',
+    'CudaArray',
+    'CudaDeviceSynchronize',
+    'CudaInternalVar',
+    'CudaThreadIdx',
+    'CudaBlockDim',
+    'CudaBlockIdx',
+    'CudaGridDim'
+)
+
+#==============================================================================
+class CudaMemCopy():
+    """Represents a call to  cuda malloc for code generation.
```

Comment:
`cuda malloc`? this does not represent memcpy?

Reply:
that is a typo it should be `cudamemcpy`.
the class does nothing now will delete it and add it in another commit.